lib/spack/spack/cmd/checkbinaries.py

```diff
@@ -0,0 +1,247 @@
+
+    range = "{0}...".format(args.base)
+
+    git_args = [
+        # Add changed files committed since branching off of develop
+        ['diff', '--name-only', '--diff-filter=ACMR', range],
+        # Add changed files that have been staged but not yet committed
+        ['diff', '--name-only', '--diff-filter=ACMR', '--cached'],
+        # Add changed files that are unstaged
+        ['diff', '--name-only', '--diff-filter=ACMR'],
+    ]
+
+    changed = set()
+
+    for arg_list in git_args:
+        files = git(*arg_list, output=str).split('\n')
+
+        for f in files:
+            # Look only for modified package.py files
+            m = _pkg_match_regex.search(f)
+            if m:
+                changed.add(m.group(1))
+
+    return sorted(changed)
+
+
+def compute_local_hash(pkg_name, spec_yaml):
+    local_spec = Spec.from_yaml(spec_yaml)
+    tty.msg('local dag hash for %s: %s' % (pkg_name, local_spec.dag_hash()))
+    local_spec.concretize()
+    return local_spec.full_hash()
+
+
+def compute_remote_hash(pkg_name, spec_yaml):
+    parsed_yaml = yaml.load(spec_yaml)
+    if 'full_hash' in parsed_yaml:
+        return parsed_yaml['full_hash']
+
+    msg = 'Binary mirror yaml file is missing full_hash attribute, aborting'
+    raise SpackError(msg)
+
+
+def check_needs_rebuild(pkg_name, spec_yaml):
+    local_pkg_hash = compute_local_hash(pkg_name, spec_yaml)
+    remote_pkg_hash = compute_remote_hash(pkg_name, spec_yaml)
+
+    tty.msg('  %s' % pkg_name)
+    tty.msg('    local hash: %s' % local_pkg_hash)
+    tty.msg('    remote hash: %s' % remote_pkg_hash)
+
+    if local_pkg_hash != remote_pkg_hash:
+        return True
+
+    return False
+
+
+def parse_yaml_list(index_html_contents):
+    result = []
+    match_list = _pkg_yaml_match_regex.findall(index_html_contents)
+    if match_list:
+        result = match_list
+    return result
+
+
+def get_mirror_rebuilds(mirror_name, mirror_url, pkg_change_set):
+    tty.msg('Checking for built specs on %s' % mirror_name)
+
+    # First fetch the index.html
+    index_path = os.path.join(mirror_url, _buildcache_path, 'index.html')
+    index_contents = None
+
+    try:
+        url = urlopen(index_path)
+    except URLError as urlErr:
+        tty.error('Unable to open url %s' % index_path)
+        tty.error(urlErr)
+        return None
+    except Exception as expn:
+        tty.error('Error getting index from mirror')
+        tty.error(expn)
+        return None
+
+    index_contents = url.read()
+
+    if not index_contents:
+        tty.error('Unable to retrieve index from mirror')
+        return None
+
+    yaml_list = parse_yaml_list(index_contents)
+
+    rebuild_list = []
+
+    for pkg_name in pkg_change_set:
+        tty.msg('yamls associated with %s' % pkg_name)
+
+        for pkg_yaml in yaml_list:
+            try:
+                pkg_yaml.index('-%s-' % pkg_name)
+                tty.msg('  %s' % pkg_yaml)
+            except Exception as expn:
```

Comment:
maybe a warning here?  Dunno why this would happen.

Reply:
At the time I wasn't quite sure that I could count on the form of the file name.  Just a warning sounds fine though.